aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/CreateHandler.java

```diff
@@ -66,11 +67,16 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
         final Collection<DBInstanceRole> desiredRoles = model.getAssociatedRoles();
 
         if (StringUtils.isNullOrEmpty(model.getDBInstanceIdentifier())) {
-            model.setDBInstanceIdentifier(instanceIdentifierFactory.newIdentifier()
-                    .withStackId(request.getStackId())
-                    .withResourceId(request.getLogicalResourceIdentifier())
-                    .withRequestToken(request.getClientRequestToken())
-                    .toString());
+            if (!StringUtils.isNullOrEmpty(model.getTargetDBInstanceIdentifier())) {
```

Comment:
Please do not use TargetDBInstanceIdentifier property to set DBInstanceIdentifier. This is a recipe for PrimaryIdentifier immutability problems.